assets/js/components/surveys/CurrentSurvey.js

```diff
@@ -16,32 +16,173 @@
+			sendSurveyEvent( 'survey_shown' );
+			setHasSentSurveyShownEvent( true );
+		}
+	}, [ questions, hasSentSurveyShownEvent, sendSurveyEvent ] );
+
+	const completions = useSelect( ( select ) => select( CORE_USER ).getCurrentSurveyCompletions() );
+	const questions = useSelect( ( select ) => select( CORE_USER ).getCurrentSurveyQuestions() );
+	const surveySession = useSelect( ( select ) => select( CORE_USER ).getCurrentSurveySession() );
+
+	const formName = surveySession ? `survey-${ surveySession.session_id }` : null;
+
+	const answers = useSelect( ( select ) => select( CORE_FORMS ).getValue( formName, 'answers' ) );
+
+	const currentQuestionOrdinal = answers?.length ? answers.reduce( ( a, b ) => {
+		return ( a.question_ordinal > b.question_ordinal ? a.question_ordinal : b.question_ordinal );
+	} ).question_ordinal + 1 : 1;
+
+	const shouldHide = useSelect( ( select ) => select( CORE_FORMS ).getValue( formName, 'hideSurvey' ) );
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	const { sendSurveyEvent } = useDispatch( CORE_USER );
+
+	const currentQuestion = questions?.find( ( question ) => {
+		return question.question_ordinal === currentQuestionOrdinal;
+	} );
+
+	const answerQuestion = useCallback( ( answer ) => {
+		// eslint-disable-next-line camelcase
+		if ( currentQuestion?.question_type === 'rating' ) {
+			sendSurveyEvent( 'question_answered', {
+				// eslint-disable-next-line camelcase
+				question_ordinal: currentQuestion?.question_ordinal,
+				answer: {
+					answer_ordinal: answer.answer_ordinal,
+				},
+			} );
+
+			setValues( formName, {
+				answers: [
+					...answers || [],
+					{
+					// eslint-disable-next-line camelcase
+						question_ordinal: currentQuestion?.question_ordinal,
+						answer_ordinal: answer.answer_ordinal,
+					},
+				],
+			} );
+		}
+	}, [ answers, currentQuestion, formName, sendSurveyEvent, setValues ] );
+
+	const ctaOnClick = useCallback( () => {
+		sendSurveyEvent( 'follow_up_link_clicked' );
+		sendSurveyEvent( 'survey_closed' );
+
+		setValues( formName, { hideSurvey: true } );
+	}, [ formName, sendSurveyEvent, setValues ] );
+
+	const dismissSurvey = useCallback( () => {
+		sendSurveyEvent( 'survey_closed' );
 
-	if ( ! usingProxy || ! currentSurvey ) {
+		setValues( formName, { hideSurvey: true } );
+	}, [ formName, sendSurveyEvent, setValues ] );
+
+	// Check to see if a completion trigger has been met.
+	let triggeredCompletion;
+	// We only have trigger conditions for questions that are answered with
+	// ordinal values right now.
+	const ordinalAnswerMap = answers?.length ? answers.reduce( ( acc, answer ) => {
+		return {
+			...acc,
+			[ answer.question_ordinal ]: answer.answer_ordinal,
+		};
+	}, {} ) : [];
+
+	// Use Array.some to avoid looping through all completions; once a matching
+	// completion has been found, treat the survey as complete.
+	completions?.some( ( completion ) => {
+		completion.trigger_condition.some( ( condition ) => {
```

Comment:
It seems like there is a missing condition here in that all questions need to be answered before completion logic is checked. I think you mentioned "completing early" in the description but as far as I know, that isn't a requirement.

Reply:
Yes that is correct. "Early completion" should not be possible. The completion data is always present in the API response, but it should only be checked / possibly displayed once all questions have been answered.

Reply:
Ahhhh, okay, that wasn't clear to me because of the `question_ordinal` attribute in the `trigger_condition` object.

My assumption in reading that was that the completion was triggered _when_ that question was answered with a particular value. But I understand now it means that "if question 1 in a three-question survey is answered with X, show this completion".

Sounds good; I'll adjust the tests and the code accordingly 👍🏻